src/containers/AccountModal/forms/ChooseOffer.tsx

```diff
@@ -12,50 +12,98 @@ import { getOffer } from '#src/services/checkout.service';
 import LoadingOverlay from '#src/components/LoadingOverlay/LoadingOverlay';
 import ChooseOfferForm from '#src/components/ChooseOfferForm/ChooseOfferForm';
 import useForm, { UseFormOnSubmitHandler } from '#src/hooks/useForm';
-import type { ChooseOfferFormData, OfferPeriodicity } from '#types/account';
+import type { ChooseOfferFormData, OfferPeriodicity, OfferType } from '#types/account';
+import type { Offer } from '#types/checkout';
 
 const ChooseOffer = () => {
   const history = useHistory();
   const { t } = useTranslation('account');
-  const { config, accessModel } = useConfigStore(({ config, accessModel }) => ({ config, accessModel }), shallow);
+  const { config } = useConfigStore(({ config }) => ({ config }), shallow);
   const { cleengSandbox, json } = config;
-  const hasOffer = accessModel === 'SVOD';
-  const { offer, setOffer } = useCheckoutStore(({ offer, setOffer }) => ({ offer, setOffer }), shallow);
+  const { setOffer, setOfferType } = useCheckoutStore(({ setOffer, setOfferType }) => ({ setOffer, setOfferType }), shallow);
 
   const cleengMonthlyOffer = json?.cleengMonthlyOffer as string;
   const cleengYearlyOffer = json?.cleengYearlyOffer as string;
 
+  const { requestedMediaOffers } = useCheckoutStore(({ requestedMediaOffers }) => ({ requestedMediaOffers: requestedMediaOffers || [] }));
+
   // `useQueries` is not strongly typed :-(
-  const { data: monthlyOfferData } = useQuery(['offer', cleengMonthlyOffer], () => getOffer({ offerId: cleengMonthlyOffer }, cleengSandbox));
-  const { data: yearlyOfferData } = useQuery(['offer', cleengYearlyOffer], () => getOffer({ offerId: cleengYearlyOffer }, cleengSandbox));
+  const { data: monthlyOfferData, isLoading: isMonthlyOfferLoading } = useQuery(['offer', cleengMonthlyOffer], () =>
+    getOffer({ offerId: cleengMonthlyOffer }, cleengSandbox),
+  );
+  const { data: yearlyOfferData, isLoading: isYearlyOfferLoading } = useQuery(['offer', cleengYearlyOffer], () =>
+    getOffer({ offerId: cleengYearlyOffer }, cleengSandbox),
+  );
+
+  const tvodOfferQueries = useQueries(
+    requestedMediaOffers?.map(({ offerId }) => ({
+      queryKey: ['requestedMediaOffer', offerId],
```

Comment:
I would keep the queryKey predictable to make it easier to invalidate (if we eventually decide to do so). Since it is getting an offer, I suggest to keep it the same e.g.

```suggestion
      queryKey: ['offer', offerId],
```

Reply:
Good point!